opencti-platform/opencti-front/src/private/components/analyses/notes/AddNotesFunctionalComponent.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { FunctionComponent, useState } from 'react';
+import { Button, Dialog, DialogContent, DialogTitle, IconButton, List, ListItem, ListItemIcon, ListItemText, Skeleton, ThemeProvider, createTheme } from '@mui/material';
+import { Add } from '@mui/icons-material';
+import Drawer from '@components/common/drawer/Drawer';
+import { useFormatter } from 'src/components/i18n';
+import SearchInput from 'src/components/SearchInput';
+import { RecordSourceSelectorProxy } from 'relay-runtime';
+import { QueryRenderer } from 'src/relay/environment';
+import { insertNode } from 'src/utils/store';
+import AddNotesLines, { addNotesLinesQuery } from './AddNotesLines';
+import { AddNotesLinesQuery$data } from './__generated__/AddNotesLinesQuery.graphql';
+import { NoteCreationForm } from './NoteCreation';
+import { NotesLinesPaginationQuery$variables } from './__generated__/NotesLinesPaginationQuery.graphql';
+import { StixCoreObjectOrStixCoreRelationshipNotesCards_data$data } from './__generated__/StixCoreObjectOrStixCoreRelationshipNotesCards_data.graphql';
+
+interface AddNotesFunctionalComponentProps {
+  stixCoreObjectOrStixCoreRelationshipId: string,
+  stixCoreObjectOrStixCoreRelationshipNotes: StixCoreObjectOrStixCoreRelationshipNotesCards_data$data,
+  paginationOptions: NotesLinesPaginationQuery$variables,
+}
+
+// TODO: Rename to AddNotes and replace AddNotes.jsx
+const AddNotesFunctionalComponent: FunctionComponent<AddNotesFunctionalComponentProps> = ({
+  stixCoreObjectOrStixCoreRelationshipId,
+  stixCoreObjectOrStixCoreRelationshipNotes,
+  paginationOptions,
+}) => {
+  const { t_i18n } = useFormatter();
+  const [open, setOpen] = useState<boolean>(false);
+  const [dialogOpen, setDialogOpen] = useState<boolean>(false);
+  const [search, setSearch] = useState<string>('');
+  const updater = (store: RecordSourceSelectorProxy, key: string) => {
+    return insertNode(store, 'Pagination_notes', { search }, key);
+  };
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setSearch('');
+  };
+
+  const handleDialogOpen = () => {
+    setDialogOpen(true);
+  };
+
+  const handleDialogClose = () => {
+    setDialogOpen(false);
+  };
+
+  const handleSearch = (keyword: string) => {
+    setSearch(keyword);
+  };
+
+  const CustomIconButtonTheme = createTheme({
+    components: {
+      MuiIconButton: {
+        styleOverrides: {
+          root: {
+            float: 'right',
+            marginTop: -15,
+          },
+        },
+      },
+    },
+  });
```

Comment:
Why using a ThemeProvider here? You can directly put the `float` and `marginTop` CSS rules in `style` property of `IconButton`